### PR TITLE
raise error in RuntimeManager.captureResults on unsuccessful deploy

### PR DESCRIPTION
**Pull request by odeac**
## Overview
Without this change the errors are silent and can go unnoticed

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3261

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by lukasz-golebiewski** [comment]
It's sad that no test had to be adapted. Do you think you could write a fresh one?

**Comment by ArturGajowy** [comment]
Don't you have to handle this error somewhere in casper as well? Did you run casper tests?

It should be the case that such an error during a propose doesn't terminate the propose. Rather, the failed deploys should be propagated for other nodes to know (or ignored, if we don't do the propagation yet).

**Comment by ArturGajowy** [comment]
Also, I've observed that RhoSpec doesn't report rholang evaluation errors neither in test nor in production code. Is this change intended to fix that?

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 1]
```diff
@@ -71,12 +71,13 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
...
-          result <- if (processedDeploy.status.isFailed) Seq.empty[Datum[ListParWithRandom]].pure[F]
-                   else {
-                     val r: F[Seq[Datum[ListParWithRandom]]] =
-                       runtime.space.getData(name).map(_.toSeq)
-                     r
-                   }
+          result <- if (processedDeploy.status.isFailed)
+                      Sync[F].raiseError(new Exception("Evaluation error:" + processedDeploy.status))
+                    else {
+                      val r: F[Seq[Datum[ListParWithRandom]]] =
```
Comment (tymm) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: Not related to your change but looks like this is a leftover from debugging and can be removed.

**Comment by odeac** [comment]
the rebase is too dificult.If we decide to do this again I'll redoit

**Pull request closed without merging**